lms/static/js/ccx/collection/schedule_collection.js

```diff
@@ -0,0 +1,269 @@
+var define = window.define || RequireJS.define;  // jshint ignore:line
+
+define(
+  [
+    'backbone',
+    'underscore',
+    'gettext',
+    'moment',
+    'js/ccx/model/schedule_model'
+  ],
+  function (
+    Backbone, _, gettext, moment,  ccxScheduleModel
+  ) {
+    'use strict';
+    return Backbone.Collection.extend({
+      model: ccxScheduleModel,
+      url: 'ccx_schedule',
+
+      showAllUnitsInScheduleTree: function() {
+        // show all units i.e chapters, subsections and verticals
+        // in schedule tree
+        var scheduleJson = this.toJSON();
+        this.changeVisibilityOfUnitInSchedule(
+          scheduleJson, this.show
+        );
+        this.reset(scheduleJson);
+      },
+
+      hideAllUnitFromScheduleTree: function() {
+        // clear schedule tree on remove all click
+        var scheduleJson = this.toJSON();
+        this.changeVisibilityOfUnitInSchedule(
+          scheduleJson, this.hide
+        );
+        this.reset(scheduleJson);
+      },
+
+      hideUnitFromScheduleTree: function(chapterLocation, sequentialLocation,
+                                         verticalLocation) {
+        // hide child (can be chapter, sequential or vertical) in collection.
+        var scheduleJson = this.toJSON();
+        var unit = this.findUnit(
+          scheduleJson,
+          chapterLocation,
+          sequentialLocation,
+          verticalLocation
+        );
+
+        if (unit) {
+          this.changeVisibilityOfUnitInSchedule(
+            [unit],
+            this.hide
+          );
+          this.reset(scheduleJson);
+        }
+      },
+
+      filterTreeData: function() {
+        return this.pruned(this.toJSON(), this.treeFilter);
+      },
+
+      filterFormData: function() {
+        return this.pruned(this.toJSON(), this.formFilter);
+      },
+
+      showUnitInScheduleTree: function(
+        chapterLocation,
+        sequentialLocation,
+        verticalLocation,
+        startDate,
+        dueDate
+      ) {
+        // hide child (can be chapter, sequential or vertical) in collection.
+        var scheduleJson = this.toJSON();
+        var errorMessage;
+
+        var units = this.findLineage(
+          scheduleJson,
+          chapterLocation,
+          sequentialLocation,
+          verticalLocation
+        );
+
+        if (units) {
+          errorMessage = this.validUnitDates(startDate, dueDate);
```

Comment:
If I sent an invalid due date (for instance, a date in the year 1920 for a course), on my sandbox I'm not seeing any error message. The date just appears to be ignored.

![image](https://cloud.githubusercontent.com/assets/484484/14290366/caaa7e16-fb2c-11e5-833c-c674b76caed2.png)